addons/point_of_sale/models/res_config_settings.py

```diff
@@ -1,21 +1,316 @@
 # -*- coding: utf-8 -*-
 
-from odoo import fields, models
+from odoo import api, fields, models
 
 
 class ResConfigSettings(models.TransientModel):
+    """
+    NOTES
+    1. Fields with 'pos' attributes are removed from the vals before super call to `create`.
+       Values of these fields are written to `pos_config_id` record after the super call.
+       So, if we want to make an atomic write to some set of fields, we need to redefine those
+       fields to have compute, readonly=False, store=True, and pos='<pos.config.field.name>'.
+    2. During `creation` of this record, each related field is written to the source record
+       *one after the other*, so constraints on the source record that are based on multiple
+       fields might not work properly. However, only the *modified* related fields are written
+       to the source field. But the identification of modified fields happen during the super
+       call, not before `create` is called. Because of this, vals contains a lot of field before
+       super call, then the number of fields is reduced after.
+    """
     _inherit = 'res.config.settings'
 
+    def _valid_field_parameter(self, field, name):
+        """Introduce 'pos' attribute to allow atomic write for the destination pos.config record."""
+        return name == 'pos' or super()._valid_field_parameter(field, name)
```

Comment:
Instead of using "pos=" on the fields, couldn we do something like this in the 'write':
```
for field_name, field in self._fields.items()
    if field.related_field and field.related_field.model_name == 'pos.config' and field_name in vals
```

It has been done here: https://github.com/odoo/odoo/blob/15.0/addons/hr/models/res_users.py#L198

Reply:
Early in the development, I started with your suggestion and to gain "user interactivity" in the form, I moved the "onchanges" from "pos_config.py" to "res_config_settings.py".

But there was an issue (can't really remember, probably hidden in the old commits). Then we decided to use "compute" for user form interactivity. However, I can't put a "compute" in a "related" field. Without "related", I can't identify whether the field is related to "pos_config_id" or not, so I decided to add a new "pos" attribute. 

What is happening now in the res.config.settings is this:

1. Form ui opens, no orm methods is triggered yet (AFAIK).
2. The user interacts with the form. `compute` methods are triggered, results are automatically reflected in the UI.
3. The user clicks save.
    * each related field is "written" to the source record (pos_config_id). (non-atomic write)
    * "pos" fields (those with compute and "pos" attributes) are collected.
    * super().create() is called without the "pos" fields.
    * collected pos fields are written to pos_config_id at once. (atomic write)

Do you think we can still improve it?

Reply:
I thought of basing it on the field name, for example, if it's `pos_receipt_header`, we'll infer that the destination field to pos.config is `receipt_header`. However, if somebody outside pos decided to introduce a field with name that starts with `pos_`, then it'll cause issue to us.